tests/unittests/PartitionerTest.cpp

```diff
@@ -175,67 +184,77 @@ TEST_F(PartitionerTest, Basic1) {
+    Node *L = F_->createFullyConnected("left_fc1", input, w2, b2);
+    L = F_->createSigmoid("left_sigmoid1", L);
+    auto *w3 = mod->createConstant(ElemKind::FloatTy, {16, 8}, "w3");
+    auto *b3 = mod->createConstant(ElemKind::FloatTy, {8}, "b3");
+    w3->getHandle<>().randomize(-range, range, mod->getPRNG());
+    b3->getHandle<>().randomize(-range, range, mod->getPRNG());
+    L = F_->createFullyConnected("left_fc2", L, w3, b3);
+    L = F_->createSigmoid("left_sigmoid2", L);
+
+    // Right branch.
+    auto *w4 = mod->createConstant(ElemKind::FloatTy, {16, 16}, "w4");
+    auto *b4 = mod->createConstant(ElemKind::FloatTy, {16}, "b4");
+    w4->getHandle<>().randomize(-range, range, mod->getPRNG());
+    b4->getHandle<>().randomize(-range, range, mod->getPRNG());
+    Node *R = F_->createFullyConnected("right_fc1", input1, w4, b4);
+    R = F_->createSigmoid("right_sigmoid1", R);
+    auto *w5 = mod->createConstant(ElemKind::FloatTy, {16, 8}, "w5");
+    auto *b5 = mod->createConstant(ElemKind::FloatTy, {8}, "b5");
+    w5->getHandle<>().randomize(-range, range, mod->getPRNG());
+    b5->getHandle<>().randomize(-range, range, mod->getPRNG());
+    R = F_->createFullyConnected("right_fc2", R, w5, b5);
+    R = F_->createSigmoid("right_sigmoid2", R);
+
+    // Join branches.
+    auto *mul = F_->createMul("mul", L, R);
+    F_->createSave("ret", mul);
+  }
 
   // Infer using the un-partitioned graph.
   Tensor in(ElemKind::FloatTy, {1, 16});
-  in.getHandle<>().randomize(-range, range, mod_.getPRNG());
-  ExecutionEngine EE;
+  in.getHandle<>().randomize(-range, range, EER.getModule().getPRNG());
+  EER.compile(CompilationMode::Infer);
+  bindings_.clear();
+  bindings_.allocate(EER.getModule().getPlaceholders());
+  updateInputPlaceholders2(bindings_,
+                           {bindings_.getPlaceholderByName("input"),
+                            bindings_.getPlaceholderByName("input1")},
+                           {&in, &in});
+  EER.run(bindings_);
+  Tensor ref = bindings_.get(bindings_.getPlaceholderByName("ret"))->clone();
 
-  EE.compile(CompilationMode::Infer, F_);
-  updateInputPlaceholders(bindings_, {input, input1}, {&in, &in});
-  EE.run(bindings_);
-  Tensor ref = res.clone();
   std::vector<DeviceInfo> devices = {{2048, "Interpreter"},
                                      {2048, "Interpreter"},
                                      {2048, "Interpreter"},
                                      {2048, "Interpreter"}};
-  Partitioner myPartitioner(&mod_, devices, /* saturateHost */ true);
+  Partitioner myPartitioner(&EEP.getModule(), devices, /* saturateHost */ true);
   CompilationContext cctx;
   auto err = myPartitioner.Partition(cctx);
   EXPECT_FALSE(errToBool(std::move(err)));
   DAGListTy dagList = std::move(myPartitioner.getPartitionResult());
-  EXPECT_EQ(mod_.getFunctions().size(), 2);
-  EXPECT_EQ(dagList.size(), 1);
-  ASSERT_TRUE(checkSaveNode(mod_));
+  ASSERT_EQ(EEP.getModule().getFunctions().size(), 2);
```

Comment:
I don't think this check is "fatal". Even it fails, we can still check the dagList.size() and then the saveNode. So I still prefer "EXPECT_EQ" here. I think the last check before running the DAG (i.e. checkSaveNode(EEP.getModule())) should be ASSERT_TRUE. But the checks before that (i.e. the ones checking the dag structure) could be EXPECT_TRUE.